Library/Homebrew/install.rb

```diff
@@ -272,50 +291,35 @@ def install_formulae(
         )
 
         begin
+          fi.prelude
           fi.fetch
           fi
+        rescue CannotInstallFormulaError => e
+          ofail e.message
         rescue UnsatisfiedRequirements, DownloadError, ChecksumMismatchError => e
           ofail "#{f}: #{e}"
           nil
         end
       end.compact
 
       formula_installers.each do |fi|
-        install_formula(fi, only_deps: only_deps)
+        install_formula(fi)
         Cleanup.install_formula_clean!(fi.formula)
       end
     end
 
-    def install_formula(formula_installer, only_deps: false)
+    def install_formula(formula_installer)
       f = formula_installer.formula
 
-      formula_installer.prelude
+      formula_installer.check_installation_already_attempted
```

Comment:
Can you explain a bit more about this change?

Reply:
See above ⬆️